src/toil/jobStores/aws/jobStore.py

```diff
@@ -1047,3 +1011,75 @@ def toItem(self):
         :return: a str for the item's name and a dictionary for the item's attributes
         """
         return self.jobStoreID, self.binaryToAttributes(cPickle.dumps(self))
+
+
+def copyKeyMultipart(srcKey, dstBucketName, dstKeyName, headers=None):
+    """
+    Copies a key from a source key to a destination key in multiple parts. Note that if the
+    destination key exists it will be overwritten implicitly, and if it does not exist a new
+    key will be created. If the destination bucket does not exist an error will be raised.
+
+    :param boto.s3.key.Key srcKey: The source key to be copied from.
+    :param str dstBucketName: The name of the destination bucket for the copy.
+    :param str dstKeyName: The name of the destination key that will be created or overwritten.
+    :param dict headers: Any headers that should be passed.
+
+    :rtype: boto.s3.multipart.CompletedMultiPartUpload
+    :return: An object representing the completed upload.
+    """
+    s3 = boto.connect_s3()
+    partSize = defaultPartSize
+    totalSize = srcKey.size
+    totalParts = (totalSize / partSize) + 1 if totalSize % partSize else (totalSize / partSize)
+    exceptionList = []
+
+    def _copyPart(partIndex):
+        if exceptionList:
+            return None
+        else:
+            try:
+                startByte = partIndex * partSize
+                startByte = startByte if startByte <= totalSize else totalSize - ((partIndex - 1) * partSize)
```

Comment:
Assuming ```n``` parts, the nth part is necessarily smaller than ```partSize``` if and only if ```totalSize % partSize != 0```.

Assuming ```totalSize % partSize != 0```. Every part except the nth and final part must be of size ```partSize```, thus  ```totalSize = (n-1) * partSize + sizeof(n)```. 
```n * partSize =  (n-1) * partSize + partSize ```
since ```partSize > sizeof(n) ``` by the assumed case,
 ```n * partSize > totalSize ``` must be true in this case. 

Assuming ```totalSize % partSize == 0``` we have that ```n * partSize = totalSize ``` since ```sizeof(n)``` is necessarily ```partSize```.

Therefore ```partIndex * partSize >= totalSize``` must be true when ```partIndex = n```.